### PR TITLE
docs: add usage for personalisation list fields

### PR DESCRIPTION
**Pull request by jonathaningram**
Also update the Go reference to use an interface{} for Value instead of
a string.

### CODE REVIEW

File: docs/src/content/sending-messages.mdx [hunk 1]
```diff
@@ -34,9 +34,56 @@ gets sent out. You might use personalisation to:
...
+In this example, `((documents))` looks just like `((name))` but its intent is to
+become a nicely formatted list of documents.
+
+Notify allows you to convert this field into a list in the generated message.
+To do this, your code should pass in a list of things as the value of the
+personalisation field, instead of a string.
+
+For example, in Go you would pass a slice of strings:
+
+```go
```
Comment (andrewoh531) [inline docs/src/content/sending-messages.mdx]: Is JSON possibly a language neutral/better example? i.e.

```json
{
   "name": "Kim",
   "documents": ["Driver license", "Medicare card"]
}
```

File: docs/src/code-examples/generate-template-preview/go.mdx [hunk 2]
```diff
@@ -3,22 +3,27 @@ lang: go
...
     <Parameter required name="id" kind="string">
-      Sign in to Notify and go to the Templates page to find the template
-      ID.
+      Sign in to Notify and go to the Templates page to find the template ID.
     </Parameter>
-    <Parameter optional name="personalisation" kind="[]struct{ Key, Value string }">
+    <Parameter
+      optional
+      name="personalisation"
+      kind="[]struct{ Key string, Value interface{} }"
```
Comment (andrewoh531) [inline docs/src/code-examples/generate-template-preview/go.mdx]: Should we update the examples for the other languages as well?

File: docs/src/content/sending-messages.mdx [hunk 3]
```diff
@@ -34,9 +34,56 @@ gets sent out. You might use personalisation to:
...
+• Driver licence
+• Medicare card
+```
+
+Generated text messages will display the list inline:
+
+```text
+Dear Kim, please bring these documents to your appointment:
+
+Driver licence and Medicare card
```
Comment (andrewoh531) [inline docs/src/content/sending-messages.mdx]: If there were more than 2 items in the list, would it become `Driver license, xxx1, xxx2 and Medicare card`?

File: docs/src/code-examples/generate-template-preview/go.mdx (hunk 2 above)
Comment (jonathaningram) [reply docs/src/code-examples/generate-template-preview/go.mdx]: Yep, good call, I forgot.

File: docs/src/content/sending-messages.mdx (hunk 3 above)
Comment (jonathaningram) [reply docs/src/content/sending-messages.mdx]: Yeah. Should my example use 3 items?

File: docs/src/content/sending-messages.mdx (hunk 1 above)
Comment (jonathaningram) [reply docs/src/content/sending-messages.mdx]: You're right, that's better.

File: docs/src/code-examples/generate-template-preview/go.mdx (hunk 2 above)
Comment (jonathaningram) [reply docs/src/code-examples/generate-template-preview/go.mdx]: Going to update the other languages separately, possibly one at a time.

Comment (jonathaningram) [reply docs/src/code-examples/generate-template-preview/go.mdx]: Updated node, will do others separately.

**Pull request merged**